inspirehep/ext/jinja_filters/record.py

```diff
@@ -355,6 +355,48 @@ def epoch_to_year_format(date):
 
     @app.template_filter()
     def construct_date_format(date):
-        starting_date = time.strftime('%Y-%m-%d', time.gmtime(int(date) / 1000.))
+        starting_date = time.strftime(
+            '%Y-%m-%d', time.gmtime(int(date) / 1000.))
         ending_date = time.strftime('%Y-12-31', time.gmtime(int(date) / 1000.))
         return starting_date + '->' + ending_date
+
+    @app.template_filter()
+    def author_urls(l, separator):
+        result = []
+        for name in l:
+            if 'name' in name:
+                url = '<a href="/search?ln=en&amp;cc=HepNames&amp;p=name:'\
+                      + name['name'] + '&amp;of=hd">' + name['name'] + '</a>'
+                result.append(url)
+        return separator.join(result)
+
+    @app.template_filter()
+    def ads_links(record):
+        import re
```

Comment:
I am importing it here because it is just used for that function in the
file. Isn't it better to import it when it is only used?

2015-12-02 16:56 GMT+01:00 Jan Åge Lavik <notifications@github.com>:

> In inspirehep/ext/jinja_filters/record.py
> <https://github.com/inspirehep/inspire-next/pull/587#discussion_r46431842>
> :
>
> >          ending_date = time.strftime('%Y-12-31', time.gmtime(int(date) / 1000.))
> >          return starting_date + '->' + ending_date
> > +
> > +    @app.template_filter()
> > +    def author_urls(l, separator):
> > +        result = []
> > +        for name in l:
> > +            if 'name' in name:
> > +                url = '<a href="/search?ln=en&amp;cc=HepNames&amp;p=name:'\
> > +                      + name['name'] + '&amp;of=hd">' + name['name'] + '</a>'
> > +                result.append(url)
> > +        return separator.join(result)
> > +
> > +    @app.template_filter()
> > +    def ads_links(record):
> > +        import re
>
> You can import this at the top of the file I believe.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inspirehep/inspire-next/pull/587/files#r46431842>.
>